File: frontend/src/microservices/flight/objects/FlightCity.ts
// 本文件定义 flight 模块的 `FlightCity`，作为传输数据并提供 JSON 编解码。

import type { FlightAirport } from './FlightAirport'

export type FlightCity = {
  cityName: string
  airports: FlightAirport[]
}

export const flightCityFromJson = (json: string): FlightCity =>
  JSON.parse(json) as FlightCity

export const flightCityToJson = (value: FlightCity): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/FlightCity.scala
// FlightCity is a backend-only grouping value object that bundles airports under a city name.
// It exists so the planner can assemble richer search/suggestion results before serialization, not so the frontend can mirror internal grouping logic.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class FlightCity(
    cityName: String,
    airports: List[FlightAirport]
)

object FlightCity:
  given sourceEncoder: Encoder[FlightCity] = deriveEncoder
  given sourceDecoder: Decoder[FlightCity] = deriveDecoder

